src/Mobs/Path.h

```diff
@@ -89,10 +89,11 @@ class cPath
 
 	/* Point retrieval functions, inlined for performance. */
 	/** Returns the next point in the path. */
-	inline Vector3i GetNextPoint()
+	inline Vector3d GetNextPoint()
 	{
 		ASSERT(m_Status == ePathFinderStatus::PATH_FOUND);
-		return m_PathPoints[m_PathPoints.size() - 1 - (++m_CurrentPoint)];
+		Vector3i Point = m_PathPoints[m_PathPoints.size() - 1 - (++m_CurrentPoint)];
+		return Vector3d(Point.x + m_HalfWidth, Point.y, Point.z + m_HalfWidth);
```

Comment:
Centering the mobs is no longer Monster.cpp's responsibility and it's done here instead (adding half their width).